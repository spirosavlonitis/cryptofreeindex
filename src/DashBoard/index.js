import React from 'react';
import PropTypes from 'prop-types';
import Image from 'react-bootstrap/lib/Image'

const DashBoard = ({coinData, coinCols, navCoins, activeKey}) => {
try{		/* Deal with multiple navbar clicks */
	return(
		<div>
			{coinCols.map((coinCol, index) =>
				<div className="col-md-4" key={index+"_dashboard"} >
					{coinCol.map( coin => 
						  <div className="usd_field" key={'col'+coin.id} >
						  	 <br/>
								 <Image src={"/images/"+coin.image} className="home_price_image" />
						     <font className="home_price" id={"_"+coin.id} >
										{ coinData[navCoins[activeKey]][coin.id] }
						     </font>
						  </div>
					)}
			  </div>
			)}
		</div>
	)
}catch (e) {
	return null;
}

}

DashBoard.propTypes = {
	coinData: PropTypes.shape({
		USD: PropTypes.shape({
			USD: PropTypes.number.isRequired,
			EUR: PropTypes.number.isRequired,
			GBP: PropTypes.number.isRequired,
		  	BTC: PropTypes.number.isRequired,
		  	LTC: PropTypes.number.isRequired,
			BCH: PropTypes.number.isRequired,
			ETH: PropTypes.number.isRequired,
			ETC: PropTypes.number.isRequired,
			ZEC: PropTypes.number.isRequired,
			DASH: PropTypes.number.isRequired,
			XMR: PropTypes.number.isRequired,
			XRP: PropTypes.number.isRequired,
			REP: PropTypes.number.isRequired,
			BAT: PropTypes.number.isRequired,
			XLM: PropTypes.number.isRequired,
		}),
	}),
	coinCols: PropTypes.arrayOf(
		PropTypes.arrayOf(
			PropTypes.shape({
					id: PropTypes.string.isRequired,
					color: PropTypes.string.isRequired,
					image: PropTypes.string.isRequired,
			})
		)
	).isRequired,
	navCoins: PropTypes.array.isRequired,
	activeKey: PropTypes.number,
}

export default DashBoard;


