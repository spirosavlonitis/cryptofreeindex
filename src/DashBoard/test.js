import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DashBoard from './index.js'

describe('DashBoard', () => {
	const props = {
		activeKey: 0,
		navCoins:[
			"USD", "EUR", "GBP", "BTC", "LTC", "BCH", "ETH", 
			"ETC", "ZEC", "DASH", "XMR", "DCR"
		],
		coinData: {
		  "USD": {
		    "USD": 1, "EUR": 1.14, "GBP": 1.27, "BTC": 5606.49, "LTC": 42.22,
		    "BCH": 346.1, "ETH": 176.28, "ETC": 7.47, "ZEC": 113.57, "DASH": 133.94,
		    "XMR": 90.41,  "DCR": 33.81
  		}
		},
		coinCols: [
		  [
		    { "id": "USD", "color": "#339933", "image": "USD_Logo.png" },
		    { "id": "BTC", "color": "#f2a900", "image": "BTC_Logo.png" },
		    { "id": "ETH", "color": "#3385ff", "image": "ETH_Logo.png" },
		    { "id": "DASH","color": "#2075bc", "image": "DASH_Logo.png"}
		  ],
		  [
		    { "id": "EUR","color": "", "image": "EUR_Logo.svg" },
		    { "id": "LTC", "color": "#d3d3d3", "image": "LTC_Logo.png" },
		    { "id": "ETC", "color": "#669073", "image": "ETC_Logo.svg" },
		    { "id": "XMR", "color": "#ff6600", "image": "XMR_Logo.png" }
		  ],
		  [
		    { "id": "GBP", "color": "", "image": "GBP_Logo.png" },
		    { "id": "BCH", "color": "#4cca47", "image": "BCH_Logo.png" },
		    { "id": "ZEC", "color": "#f4b728", "image": "ZEC_Logo.svg" },
		    { "id": "XRP", "color": "#62D0C9", "image": "xrp-symbol-white-128.png" }
		  ]
		]
	}

	it('renders without crashing', () =>{
		const div = document.createElement('div');
		ReactDOM.render(<DashBoard {...props} />, div)
		ReactDOM.unmountComponentAtNode(div);
	});

	test('snapshot', () =>{
		const element = renderer.create(<DashBoard {...props} />);
		const tree = element.toJSON();
		expect(tree).toMatchSnapshot();
	});


})
