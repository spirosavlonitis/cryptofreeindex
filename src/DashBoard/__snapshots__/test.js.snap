// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashBoard snapshot 1`] = `
<div>
  <div
    className="col-md-4"
  >
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/USD_Logo.png"
      />
      <font
        className="home_price"
        id="_USD"
      >
        1
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/BTC_Logo.png"
      />
      <font
        className="home_price"
        id="_BTC"
      >
        5606.49
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/ETH_Logo.png"
      />
      <font
        className="home_price"
        id="_ETH"
      >
        176.28
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/DASH_Logo.png"
      />
      <font
        className="home_price"
        id="_DASH"
      >
        133.94
      </font>
    </div>
  </div>
  <div
    className="col-md-4"
  >
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/EUR_Logo.svg"
      />
      <font
        className="home_price"
        id="_EUR"
      >
        1.14
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/LTC_Logo.png"
      />
      <font
        className="home_price"
        id="_LTC"
      >
        42.22
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/ETC_Logo.svg"
      />
      <font
        className="home_price"
        id="_ETC"
      >
        7.47
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/XMR_Logo.png"
      />
      <font
        className="home_price"
        id="_XMR"
      >
        90.41
      </font>
    </div>
  </div>
  <div
    className="col-md-4"
  >
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/GBP_Logo.png"
      />
      <font
        className="home_price"
        id="_GBP"
      >
        1.27
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/BCH_Logo.png"
      />
      <font
        className="home_price"
        id="_BCH"
      >
        346.1
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/ZEC_Logo.svg"
      />
      <font
        className="home_price"
        id="_ZEC"
      >
        113.57
      </font>
    </div>
    <div
      className="usd_field"
    >
      <br />
      <img
        className="home_price_image"
        src="/images/DCR_Logo.png"
      />
      <font
        className="home_price"
        id="_DCR"
      >
        33.81
      </font>
    </div>
  </div>
</div>
`;
